build/build.go

```diff
@@ -584,22 +589,28 @@ func (db *DeploymentBuilder) buildPlatformNetworks(d *Deployment) error {
 				if index >= len(always_generate_networks) || always_generate_networks[index] != n.Type {
 					skip = true
 				}
-				network_type = n.Type
+				network = n
 				break
 			}
 		}
 
 		if !skip {
-			net, err := starlingxv1.NewPlatformNetwork(p.Name, db.namespace, p, network_type)
+			net, err := starlingxv1.NewPlatformNetwork(db.namespace, p, network)
+			if err != nil {
+				return err
+			}
+			// We are executing platform network filters after appending because CoreNetworkFilter works on
+			// d.PlatformNetworks rather than individual PlatformNetwork resource.
+			// We are passing pointer to PlatformNetwork resource (net), hence, calling filterPlatformNetworks
+			// before or after append wouldn't make any difference and either way it would work.
+			d.PlatformNetworks = append(d.PlatformNetworks, net)
+			err = db.filterPlatformNetworks(net, d)
 			if err != nil {
 				return err
 			}
-			nets = append(nets, net)
 		}
 	}
 
```

Comment:
if skip, we lose the d.PlatformNetworks = nets, suggest to remain the previous logic

Reply:
fixed.

Reply:
If skip = true: d.PlatformNetworks = make([]*starlingxv1.PlatformNetwork, 0)

Reply:
I see a problem if we add it under `If skip = true` condition. 
What if we have `pools=['oam', 'mgmt', 'storage' ... , 'pxeboot']`:
The last one being `pxeboot` address pool, skip would be `true` and executing `d.PlatformNetworks = make([]*starlingxv1.PlatformNetwork, 0)` in that condition would erase all the networks added so far ('oam', 'mgmt', 'storage') which is not intended.